Sync/Synchronizers/Bookmarks/BookmarksSynchronizer.swift

```diff
@@ -120,18 +120,66 @@ open class BufferingBookmarksSynchronizer: TimestampedSingleCollectionSynchroniz
 
         if !AppConstants.shouldMergeBookmarks {
             run = doMirror >>== { result in
-                // Just validate to report statistics.
-                if case .completed = result {
-                    log.debug("Validating completed buffer download.")
-                    return buffer.validate().bind { validationResult in
-                        if let invalidError = validationResult.failureValue as? BufferInvalidError {
-                            self.statsSession.validationStats = self.validationStatsFrom(error: invalidError)
+                guard case .completed = result else {
+                    return deferMaybe(result)
+                }
+                return storage.getLocalBookmarksAdditions() >>== { newBookmarks -> Success in
+                    guard newBookmarks.count > 0 else {
+                        return succeed()
+                    }
+                    // This assumes that "Mobile Bookmarks" exists in the buffer, which will be false if no computer ever connected to that sync account
+                    return buffer.getBufferItemAndChildrenWithGUID(BookmarkRoots.MobileFolderGUID) >>== { curMobileRoot in
+                        let newChildren = newBookmarks.values.map { $0.guid }
+                        var allChildren = curMobileRoot.children ?? []
+                        // We do not use a set in order to keep the children's order
+                        for child in newChildren {
+                            if !allChildren.contains(child) {
```

Comment:
Sorry, I misread a little; you're not materializing all of the child records, but you are cloning unnecessarily. We discussed further on Slack.